analyses/transcriptomic-dimension-reduction/04-explore-sequencing-center-effects.Rmd

```diff
@@ -0,0 +1,433 @@
+
+# At one center?
+n_centers_cg_full %>%
+  dplyr::filter(n == 1)
+```
+
+
+## UMAPs
+
+Here, we'll make some UMAPs to explore potential batch effects specifically in the **stranded** data, which was processed separately from the polyA data and is primarily reported on in the manuscript due to QC issues with the polyA data.
+
+First, let's prepare the data a little further:
+```{r}
+# only keep stranded for the rest and set order for `seq_center` as group size
+stranded_df <- umap_df_palette %>%
+  dplyr::filter(RNA_library == "stranded") %>%
+  dplyr::select(-RNA_library) %>%
+  dplyr::mutate(seq_center = forcats::fct_infreq(seq_center)) %>%
+  # arranging will help see points:
+  dplyr::arrange(seq_center)
+
+# set up color palettes
+bh_df <- palette_mapping_df %>%
+  dplyr::select(broad_histology_display, broad_histology_hex) %>%
+  dplyr::distinct() 
+pal_bh<- bh_df$broad_histology_hex
+names(pal_bh) <- bh_df$broad_histology_display
+
+cg_df <- palette_mapping_df %>%
+  dplyr::select(cancer_group_display, cancer_group_hex) %>%
+  dplyr::distinct() %>%
+  tidyr::drop_na()
+pal_cg<- cg_df$cancer_group_hex
+names(pal_cg) <- cg_df$cancer_group_display
+```
+
+### Diagnosis-free UMAP
+
+Without specifically looking at diagnoses, we'll first check if the centers cluster together. 
+Below, we see that the _vast_ majority of samples came from `NantOmics`, which itself reduces a lot of batch concerns. 
+Red circle points (`BGI@CHOP Genome Center`) are broadly distributed, but blue triangle points (`BGI`) are mostly in one location; but, there are so few samples from this center that it is hard to draw robust conclusions from that clustering.
+Moreover, as the plots below show, these triangle points are clustering with `NantOmics` points of the same broad histology and/or cancer group.
+
+```{r}
+ggplot(stranded_df) + 
+  aes(x = UMAP1, y = UMAP2, 
+      shape = seq_center, 
+      fill = seq_center, 
+      color = seq_center, 
+      alpha = seq_center) + 
+  geom_point() + 
+  scale_shape_manual(values = c(19, 21, 24)) + 
+  # first fill is ignored 
+  scale_fill_manual(values  = c("black", "red", "blue")) +
+  scale_color_manual(values = c("grey70", "black", "black")) +
+  scale_alpha_manual(values = c(0.4, 0.8, 0.8)) +
+  ggpubr::theme_pubr()
+```
+
+
+### Broad histology UMAP
+
+We can make this finer-grained by exploring the distributions of broad histologies within this UMAP.
+Shapes denote the sequencing center, and colors denote the broad histology.
+
+```{r}
+ggplot(stranded_df) + 
+  aes(x = UMAP1, y = UMAP2, 
+      shape = seq_center, 
+      fill = broad_histology_display, 
+      color = seq_center, 
+      alpha = seq_center) + 
+  geom_point() + 
+  scale_shape_manual(values = c(21, 22, 24)) + 
+  scale_fill_manual(values  = pal_bh) +
+  scale_color_manual(values = c("grey70", "black", "black")) +
+  scale_alpha_manual(values = c(0.4, 0.6, 0.6)) +
+  ggpubr::theme_pubr() + 
+  guides(fill = FALSE, alpha = FALSE)
```

Comment:
Can we have the `fill` guide here? I don't know my cancer types that well. Or maybe only on the plots where we reduce to histologies shared across sites?